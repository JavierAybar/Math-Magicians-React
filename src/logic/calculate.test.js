import calculate from './calculate';
describe('calculate', () => {
  test('when buttonName is AC should return a obj with null values', () => {
    const obj = { total: '7', next: '2', operation: '+' };
    const buttonName = 'AC';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('If there is an operation, update next', () => {
    const obj = { total: '7', next: '2', operation: '+' };
    const buttonName = '1';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('If there is no operation, update next and clear the value', () => {
    const obj = { total: '7', next: '2', operation: null };
    const buttonName = '3';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('when buttonName is . and next already contains ., no change should happen', () => {
    const obj = { total: null, next: '2.3', operation: null };
    const buttonName = '.';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('when buttonName is = and there is a operation, return a obj updated with the total', () => {
    const obj = { total: '9', next: '2', operation: '+' };
    const buttonName = '=';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('when buttonName is = and there is no operation, return a empty obj', () => {
    const obj = { total: null, next: '2', operation: null };
    const buttonName = '=';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('when buttonName is +/- and there is a total value, update total correctly', () => {
    const obj = { total: '7', next: null, operation: null };
    const buttonName = '+/-';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
  test('when buttonName is +/- and there are no values, return empty object', () => {
    const obj = { total: null, next: null, operation: null };
    const buttonName = '+/-';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
});