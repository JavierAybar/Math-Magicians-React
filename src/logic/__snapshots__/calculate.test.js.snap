// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate If there is an operation, update next 1`] = `
Object {
  "next": "21",
  "operation": "+",
  "total": "7",
}
`;

exports[`calculate If there is no operation, update next and clear the value 1`] = `
Object {
  "next": "23",
  "total": null,
}
`;

exports[`calculate when buttonName is +/- and there are no values, return empty object 1`] = `Object {}`;

exports[`calculate when buttonName is +/- and there is a total value, update total correctly 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "-7",
}
`;

exports[`calculate when buttonName is . and next already contains ., no change should happen 1`] = `
Object {
  "next": "2.3",
  "operation": null,
  "total": null,
}
`;

exports[`calculate when buttonName is = and there is a operation, return a obj updated with the total 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "11",
}
`;

exports[`calculate when buttonName is = and there is no operation, return a empty obj 1`] = `Object {}`;

exports[`calculate when buttonName is AC should return a obj with null values 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;
